disco_aws_automation/disco_metanetwork.py

```diff
@@ -301,21 +369,80 @@ def ip_by_offset(self, offset):
 
         return desired_address
 
-    def add_route(self, destination_cidr_block, gateway_id):
+    def add_gateway_routes(self, route_tuples):
+        """"
+        Add a list of gateway routes to all the subnets' route tables. Each route
+        is a tuple that contains 2 values:
+        new_route_tuples[0]: destination CIDR block
+        new_route_tuples[1]: gateway ID
+        """
+        for route_tuple in route_tuples:
+            self._add_gateway_route(route_tuple[0], route_tuple[1])
+
+    def delete_gateway_routes(self, dest_cidr_blocks):
+        """"
+        Delete the routes to destination CIDR blocks from all the subnets' route tables.
+        """
+        if self.centralized_route_table:
+            for dest_cidr_block in dest_cidr_blocks:
+                self._connection.delete_route(
+                    route_table_id=self.centralized_route_table.id,
+                    destination_cidr_block=dest_cidr_block
+                )
+        else:
+            for dest_cidr_block in dest_cidr_blocks:
+                for disco_subnet in self.disco_subnets.values():
+                    disco_subnet.delete_route(dest_cidr_block)
+
+    def update_gateway_routes(self, desired_route_tuples, dry_run=False):
```

Comment:
We could make the `delete_gateway_routes` method private, but the `add_gateway_routes` method is actually also used in `DiscoVPCGateways`.